datafusion/physical-plan/src/filter.rs

```diff
@@ -200,56 +200,42 @@ impl ExecutionPlan for FilterExec {
 
     /// The output statistics of a filtering operation can be estimated if the
     /// predicate's selectivity value can be determined for the incoming data.
-    fn statistics(&self) -> Statistics {
+    fn statistics(&self) -> Result<Statistics> {
         let predicate = self.predicate();
 
-        if !check_support(predicate, &self.schema()) {
-            return Statistics::default();
+        let schema = self.schema();
+        if !check_support(predicate, &schema) {
+            return Ok(Statistics::new_unknown(&schema));
         }
+        let input_stats = self.input.statistics()?;
 
-        let input_stats = self.input.statistics();
-        let input_column_stats = match input_stats.column_statistics {
-            Some(stats) => stats,
-            None => self
-                .schema()
-                .fields
-                .iter()
-                .map(|field| {
-                    ColumnStatistics::new_with_unbounded_column(field.data_type())
-                })
-                .collect::<Vec<_>>(),
-        };
-
-        let starter_ctx =
-            AnalysisContext::from_statistics(&self.input.schema(), &input_column_stats);
-
-        let analysis_ctx = match analyze(predicate, starter_ctx) {
-            Ok(ctx) => ctx,
-            Err(_) => return Statistics::default(),
-        };
+        let num_rows = input_stats.num_rows;
+        let total_byte_size = input_stats.total_byte_size;
+        let input_analysis_ctx = AnalysisContext::try_from_statistics(
+            &self.input.schema(),
+            &input_stats.column_statistics,
+        )?;
+        let analysis_ctx = analyze(predicate, input_analysis_ctx)?;
 
         let selectivity = analysis_ctx.selectivity.unwrap_or(1.0);
```

Comment:
```suggestion
        // Estimate (inexact) selectivity of predicate
        let selectivity = analysis_ctx.selectivity.unwrap_or(1.0);
```

I think this would help explain why the output estimates are `Inexact`